src/Model/common_datastructures.hpp

```diff
@@ -40,19 +40,55 @@
 #ifndef MODEL_COMMONDATASTRUCTURES_HPP_
 #define MODEL_COMMONDATASTRUCTURES_HPP_
 
+#include <Kernels/precision.hpp>
+
+
 namespace seissol {
   namespace model {
-    struct ElasticMaterial {
-      double lambda;
-      double mu;
+    enum class MaterialType {
+      elastic,
+      viscoelastic,
+      elastoplastic,
+      viscoplastic,
+      anisotropic,
+      none
+    };
+
+    //could be made abstract, as soon as the LTS-Tree supports polymorphism 
+    struct Material {
       double rho;
-    };    
+      virtual double getMaxWaveSpeed() const { return 0; };
+      virtual double getPWaveSpeed() const { return 0; };
+      virtual double getSWaveSpeed() const { return 0; };
+      virtual void getFullElasticTensor(real fullTensor[81]) const {}; 
+      virtual MaterialType getMaterialType() const { return MaterialType::none; } ;
```

Comment:
I don't like defining all these functions -> if they are every used you have no idea why stuff breaks
maybe add at least an error to all functions